### PR TITLE
Global: Move PyOpenColorIO to vendor/python 

### PR DESCRIPTION
**Pull request by boberfly**
So that DCCs don't conflict with their own.

See https://github.com/ynput/OpenPype/pull/4267#issuecomment-1537153263 for the issue with Gaffer.

I'm not sure if this is the correct approach, but I assume PySide/Shiboken is under `vendor/python` for this reason as well...

### CODE REVIEW

**Comment by boberfly** [comment]
@iLLiCiTiT alright I force-pushed with this re-based against `release/next-minor`

File: tools/fetch_thirdparty_libs.py [hunk 1]
```diff
@@ -67,40 +67,43 @@ def _print(msg: str, message_type: int = 0) -> None:
...
-                "-m", "pip", "install", "--upgrade", qtbinding_arg,
+                "-m", "pip", "install", "--upgrade", arg,
                 "-t", str(python_vendor_dir)
             ],
             check=True,
             stdout=subprocess.DEVNULL
         )
     except subprocess.CalledProcessError as e:
-        _print("Error during PySide2 installation.", 1)
+        _print("Error during %s installation." % package, 1)
```
Comment (iLLiCiTiT) [inline tools/fetch_thirdparty_libs.py:95]:
```suggestion
        _print(f"Error during {package} installation.", 1)
```

**Review by iLLiCiTiT — APPROVED** [review]
LGTM

File: tools/fetch_thirdparty_libs.py (hunk 1 above)
Comment (boberfly) [reply tools/fetch_thirdparty_libs.py]: Done

**Review by jakubjezek001 — APPROVED** [review]
LGTM ;) Thanks you!

**Review by antirotor — APPROVED** [review]

**Pull request merged**